src/coffee/directives/api/models/parent/polygons-parent-model.coffee

```diff
@@ -129,11 +130,13 @@ angular.module('uiGmapgoogle-maps.directives.api.models.parent')
             uiGmapPromise.promise( => @figureOutState @idKey, scope, @plurals, @modelKeyComparison)
             .then (state) =>
               payload = state
-              _async.each payload.removals, (id) =>
-                child = @plurals.get(id)
+              if(payload.updates.length)
+                #TODO: not supporting updates yet
+                $log.warning("polygons updates: #{payload.updates.length} will be missed")
+              _async.each payload.removals, (child) =>
                 if child?
                   child.destroy()
-                  @plurals.remove(id)
```

Comment:
id was not being passed in to iterator. If you look at markers it is now a childModel. So this needs to be fixed in PolylinesParentModel as well. I will just copy all of these changes into it afterwards or make a base class.